### ISSUE TITLE
False positive with unwrap and prefer-lodash-method?

### ISSUE THREAD

**Issue by melwas** [wix/eslint-plugin-lodash#81]
Hi, I have a question regarding the snippet below:

```
function filterAssets(assets, type, provider) {
    var query = _.chain(assets).values().filter(['type', type]);  // lodash/unwrap
    query = provider ? query.filter(['provider', provider]) : query; // lodash/prefer-lodash-method
    return query.value();
}
```

The plugin reports `lodash/unwrap` on line 2, and `lodash/prefer-lodash-method` on line 3. I suppose it is false positive and plugin should not report this?

**Comment by ganimomer** [comment]
Hi,
Both rules are technically working correctly. The `unwrap` rule checks that the chain is unwrapped in the same statement, and not eventually, and `prefer-lodash-method` doesn't check the case where the value is a Lodash wrapper from a previous statement.
Both checks would make both rules a whole lot more complicated and probably significantly slower.
In our internal use of the plugin, on the rare cases we need split chains, we turn off the specific rule for the specific line with a comment.
Another way to avoid the rules from reporting an error in this case would be:
```js
function filterAssets(assets, type, provider) {
    var query = _.chain(assets)
      .values()
      .filter(['type', type]);
      .filter(provider ? ['provider', provider] : _.stubTrue)
      .value();
}
```
Of course this is just a suggestion.
